x/emissions/abci_test.go

```diff
@@ -1,229 +1,180 @@
-//	file, err := filepath.Abs(fp)
-//	if err != nil {
-//
-//		return nil, err
-//	}
-//	file = filepath.Clean(file)
-//	input, err := ioutil.ReadFile(file) // #nosec G304
-//	if err != nil {
-//		return nil, err
-//	}
-//	err = json.Unmarshal(input, &data)
-//	if err != nil {
-//		return nil, err
-//	}
-//	formatedData := make([]EmissionTestData, len(data))
-//	for i, dd := range data {
-//		fl, err := strconv.ParseFloat(dd.DurationFactor, 64)
-//		if err != nil {
-//			return nil, err
-//		}
-//		dd.DurationFactor = fmt.Sprintf("%0.18f", fl)
-//		formatedData[i] = dd
-//	}
-//	return formatedData, nil
-//}
-//
-//func GenerateTestDataMaths(app *zetaapp.App, ctx sdk.Context, testMaxHeight int64, fileName string) {
-//	var generatedTestData []EmissionTestData
-//	reserverCoins := app.BankKeeper.GetBalance(ctx, emissionsModuleTypes.EmissionsModuleAddress, config.BaseDenom)
-//	startHeight := ctx.BlockHeight()
-//	for i := startHeight; i < testMaxHeight; i++ {
-//		reservesFactor := sdk.NewDecFromInt(reserverCoins.Amount)
-//		bondFactor := app.EmissionsKeeper.GetBondFactor(ctx, app.StakingKeeper)
-//		durationFactor := app.EmissionsKeeper.GetDurationFactor(ctx)
-//		blockRewards := reservesFactor.Mul(bondFactor).Mul(durationFactor)
-//		generatedTestData = append(generatedTestData, EmissionTestData{
-//			BlockHeight:    i,
-//			BondFactor:     bondFactor,
-//			DurationFactor: durationFactor.String(),
-//			ReservesFactor: reservesFactor,
-//		})
-//		validatorRewards := sdk.MustNewDecFromStr(app.EmissionsKeeper.GetParams(ctx).ValidatorEmissionPercentage).Mul(blockRewards).TruncateInt()
-//		observerRewards := sdk.MustNewDecFromStr(app.EmissionsKeeper.GetParams(ctx).ObserverEmissionPercentage).Mul(blockRewards).TruncateInt()
-//		tssSignerRewards := sdk.MustNewDecFromStr(app.EmissionsKeeper.GetParams(ctx).TssSignerEmissionPercentage).Mul(blockRewards).TruncateInt()
-//		truncatedRewards := validatorRewards.Add(observerRewards).Add(tssSignerRewards)
-//		reserverCoins = reserverCoins.Sub(sdk.NewCoin(config.BaseDenom, truncatedRewards))
-//		ctx = ctx.WithBlockHeight(i + 1)
-//	}
-//	GenerateSampleFile(fileName, generatedTestData)
-//}
-//
-//func GenerateSampleFile(fp string, data []EmissionTestData) {
-//	file, _ := json.MarshalIndent(data, "", " ")
-//	_ = ioutil.WriteFile(fp, file, 0600)
-//}
+import (
+	"math/rand"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/zetacore/cmd/zetacored/config"
+	"github.com/zeta-chain/zetacore/common"
+	keepertest "github.com/zeta-chain/zetacore/testutil/keeper"
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	emissionsModule "github.com/zeta-chain/zetacore/x/emissions"
+	emissionstypes "github.com/zeta-chain/zetacore/x/emissions/types"
+	observerTypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestBeginBlocker(t *testing.T) {
```

Comment:
Can we add a positive case test?
The emissions pool is decreasing and withdrawable emissions/validator rewards are emitted.